EngineLayer/CrosslinkSearch/CrosslinkSearchEngine.cs

```diff
@@ -515,14 +606,21 @@ private List<MatchedFragmentIon> ScoreChildScan(Ms2ScanWithSpecificMass parentSc
                     mainPeptide.OneBasedStartResidueInProtein, mainPeptide.OneBasedEndResidueInProtein,
                     mainPeptide.CleavageSpecificityForFdrCategory, mainPeptide.PeptideDescription,
                     mainPeptide.MissedCleavages, mod, mainPeptide.NumFixedMods);
-
-                peptideWithMod.Fragment(CommonParameters.ChildScanDissociationType, FragmentationTerminus.Both, childProducts);
+          
+                    peptideWithMod.Fragment(CommonParameters.MS3ChildScanDissociationType, FragmentationTerminus.Both, childProducts);              
             }
-            else if (Math.Abs(childScan.PrecursorMass - parentScan.PrecursorMass) < 0.01 && CommonParameters.DissociationType != CommonParameters.ChildScanDissociationType)
+            else if (Crosslinker.Cleavable && Math.Abs(childScan.PrecursorMass - parentScan.PrecursorMass) < 0.01)
             {
-                // same species got fragmented twice, the second time with a different dissociation type
-                childProducts = CrosslinkedPeptide.XlGetTheoreticalFragments(CommonParameters.ChildScanDissociationType,
-                    Crosslinker, new List<int> { possibleSite }, otherPeptide.MonoisotopicMass, mainPeptide).First().Item2;
+                if (childScan.TheScan.MsnOrder == 2)
+                {
+                    childProducts = CrosslinkedPeptide.XlGetTheoreticalFragments(CommonParameters.MS2ChildScanDissociationType,
+                        Crosslinker, new List<int> { possibleSite }, otherPeptide.MonoisotopicMass, mainPeptide).First().Item2;
+                }
+                else
+                {
+                    childProducts = CrosslinkedPeptide.XlGetTheoreticalFragments(CommonParameters.MS3ChildScanDissociationType,
```

Comment:
unit test?

Reply:
This is for a rare weird case. I prefer not to test it.
It tried to corver the situation that the MS3 scan have the same precursor mass as the MS2.  
This is so rare that I only see it in one data. In order to test it, I need to add the data in MM and just make sure the code run it through.